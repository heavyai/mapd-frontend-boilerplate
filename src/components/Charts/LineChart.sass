.wrapper
  width: 100%
  height: 100%
  display: flex
  align-items: center
  justify-content: center

.container
  width: 800px
  height: 400px
