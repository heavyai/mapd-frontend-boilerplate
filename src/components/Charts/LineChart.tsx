import React from 'react'
import 'mapd3/dist/mapd3.css'
import { Chart, DataManager } from 'mapd3'

const styles = require( './LineChart.sass' )


// time, number, string
const keyType = 'time'
const dataConfig = {
  keyType,
  range: [ 0, 100 ],
  pointCount: 100,
  groupCount: 1,
  lineCount: 1,
  stringMinMaxLength: [ 5, 10 ],
  randomStepSize: 50,
  nullRatio: 10
}
const dataManager = DataManager()
  .setConfig( dataConfig )
const data = dataManager.generateTestDataset()

const chartConfig = {
  keyType: 'time',
  chartType: 'line',
  tooltipIsEnabled: false,
  legendIsEnabled: false,
  binningIsEnabled: false,
  xDomainEditorIsEnabled: false,
  yDomainEditorIsEnabled: false,
  brushRangeIsEnabled: false,
  brushIsEnabled: true
}


type Props = {}

const LineChart: React.StatelessComponent<Props> = () => {
  const initializeChart = ( ref: HTMLDivElement ) => {
    Chart( ref )
      .setConfig( chartConfig )
      .setData( data )
  }

  return (
    <div className={ styles.wrapper }>
      <div className={ styles.container } ref={ initializeChart } />
    </div>
  )
}


export default LineChart
