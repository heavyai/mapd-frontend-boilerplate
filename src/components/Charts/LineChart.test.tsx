import React from 'react'
import { render } from 'enzyme'

import LineChart from './LineChart'


test( 'Component renders as expected', () => {
  const component = render(
    <LineChart />
  )

  expect( component ).toMatchSnapshot()
} )
