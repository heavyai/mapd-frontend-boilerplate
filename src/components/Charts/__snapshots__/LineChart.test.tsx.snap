// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component renders as expected 1`] = `
<div
  class="wrapper"
>
  <div
    class="container"
  />
</div>
`;
